Sources/Fuzzilli/Lifting/JavaScriptLifter.swift

```diff
@@ -283,12 +286,11 @@ public class JavaScriptLifter: Lifter {
             case let op as BeginPlainFunctionDefinition:
                 liftFunctionDefinitionBegin(op, "function")
 
-            case let op as BeginStrictFunctionDefinition:
-                liftFunctionDefinitionBegin(op, "function")
-                w.emit("'use strict';")
-
             case let op as BeginArrowFunctionDefinition:
                 let params = liftFunctionDefinitionParameters(op)
+                if op.isStrict {
```

Comment:
Isn't this before the ArrowFunction now? Shouldn't it be after line 295?

Reply:
The few examples that I've come across had the strict directive before the arrow function definition. Let me look this up again

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#arrow_functions_used_as_methods

It appears that strict and non-strict variants of arrow functions are semantically the same. I suppose we could emit the directive in the block rather than before it.

Reply:
So I think it has to be inside the block:
```javascript
> let nonstrict = () => { no_such_var = 42; }
undefined
> let strict = () => { 'use strict'; no_such_var2 = 42; }
undefined
> nonstrict()
undefined
> strict()
Uncaught ReferenceError: no_such_var2 is not defined
    at strict (REPL2:1:49)
> 
```
It's also how I'd interpret "Function code is strict mode code if the associated FunctionDeclaration, FunctionExpression, GeneratorDeclaration, GeneratorExpression, AsyncFunctionDeclaration, AsyncFunctionExpression, AsyncGeneratorDeclaration, AsyncGeneratorExpression, MethodDefinition, ArrowFunction, or AsyncArrowFunction is contained in strict mode code or if the code that produces the value of the function's [[ECMAScriptCode]] internal slot begins with a Directive Prologue that contains a Use Strict Directive." (from https://tc39.es/ecma262/#sec-strict-mode-code)